### PR TITLE
dependencies: `azurerm_service_plan` - update to use `hashicorp/go-azure-sdk`

### PR DESCRIPTION
**Pull request by jackofallops**
Dependent on: https://github.com/hashicorp/terraform-provider-azurerm/pull/24485

### CODE REVIEW

File: internal/services/appservice/linux_function_app_resource.go [hunk 1]
```diff
@@ -1199,16 +1203,16 @@ func (r LinuxFunctionAppResource) CustomImporter() sdk.ResourceRunFunc {
 		if props.ServerFarmID == nil {
 			return fmt.Errorf("determining Service Plan ID for Linux %s: %+v", id, err)
 		}
-		servicePlanId, err := parse.ServicePlanID(*props.ServerFarmID)
+		servicePlanId, err := commonids.ParseAppServicePlanIDInsensitively(*props.ServerFarmID)
```
Comment (catriona-m) [inline internal/services/appservice/linux_function_app_resource.go:1206]: just double checking it was intentional to use `ParseAppServicePlanIDInsensitively` given it wasn't case insensitive before?

**Review by catriona-m — APPROVED** [review]
LGTM!

File: internal/services/appservice/linux_function_app_resource.go (hunk 1 above)
Comment (jackofallops) [reply internal/services/appservice/linux_function_app_resource.go:1206]: Yeah, unfortunately the API always returns this in lowercase currently, so any processing of a response must be case insensitive.

**Comment by jackofallops** [comment]
Tests looking OK, transient errors fail different tests in the below runs:

image

image

Service Plan ID migrations tested based on creation via `3.86.0`.

**Pull request merged**